File: examples/strongscript/7.ts
class A {
	f() : !number { return 1 }
}

class B {
	g() : !string { return "meh" }
}

class DAB {
	f() : !number { return 2 }	
	g() : !string { return "meh" }
}

class D {
	main(x:DAB, i:!number) {
		if (i == 0) { return x.f() } else { return x.g() }
	}
}

// use case:
new D().main( new A(), 0 )
new D().main( new B(), 1 )

File: examples/reticulated_python/7.py
class A:
	def f() -> Int:
		return 1
class B:
	def g() -> String:
		return "meh"

class DAB(A,B):
	def f() -> Int:
		return 1
	def g() -> String:
		return "meh"

class D:
	def main(self:D, x:DAB, i : Int):
		if i == 0:
			return x.f() 
		else:
			return x.g()

D().main(A(), 0)
